vis-cmds.c

```diff
@@ -632,16 +632,42 @@ static bool cmd_earlier_later(Vis *vis, Win *win, Command *cmd, const char *argv
 	return pos != EPOS;
 }
 
+static size_t space_replace(char *dest, const char *src, size_t maxlen) {
+	char *to = dest;
+	const char *from = src;
+	size_t destlen = 0;
+	size_t count = 0;
+	const size_t size = strlen("␣");
+	for (; *from != '\0' && destlen < maxlen - size - 1; from++) {
+		if (*from == ' ' && count < 3) {
+			strncpy(to, "␣", size);
+			count++;
+			destlen += size;
+			to += size;
+		} else {
+			*to = *from;
+			destlen++;
+			to++;
+		}
+	}
+	*to = '\0';
+	return count;
+}
+
 static bool print_keylayout(const char *key, void *value, void *data) {
-	return text_appendf(data, "  %-18s\t%s\n", key[0] == ' ' ? "␣" : key, (char*)value);
+	char buf[25];
```

Comment:
I would also be fine with `buf` being a little bit too long, say 64 bytes,
since it not hanging around anyways.